src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+import OMEXML from './omeXML';
+import { isInTileBounds } from './utils';
+
+// Credit to https://github.com/zbjornson/node-bswap/blob/master/bswap.js for the implementation.
+// I could not get this to import, and it doesn't appear anyone else can judging by the "Used by" on github.
+function flip16(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 2);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 2) {
+    const t = flipper[i];
+    flipper[i] = flipper[i + 1];
+    flipper[i + 1] = t;
+  }
+}
+
+function flip32(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 4);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 4) {
+    let t = flipper[i];
+    flipper[i] = flipper[i + 3];
+    flipper[i + 3] = t;
+    t = flipper[i + 1];
+    flipper[i + 1] = flipper[i + 2];
+    flipper[i + 2] = t;
+  }
+}
+
+function flip64(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 8);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 8) {
+    let t = flipper[i];
+    flipper[i] = flipper[i + 7];
+    flipper[i + 7] = t;
+    t = flipper[i + 1];
+    flipper[i + 1] = flipper[i + 6];
+    flipper[i + 6] = t;
+    t = flipper[i + 2];
+    flipper[i + 2] = flipper[i + 5];
+    flipper[i + 5] = t;
+    t = flipper[i + 3];
+    flipper[i + 3] = flipper[i + 4];
+    flipper[i + 4] = t;
+  }
+}
+
+function flipEndianness(arr) {
+  switch (arr.BYTES_PER_ELEMENT) {
+    case 1:
+      // no op
+      return;
+    case 2:
+      flip16(arr);
+      break;
+    case 4:
+      flip32(arr);
+      break;
+    case 8:
+      flip64(arr);
+      break;
+    default:
+      throw new Error('Invalid input');
+  }
+}
+
+/**
+ * This class serves as a wrapper for fetching tiff data from a file server.
+ * */
+export default class OMETiffLoader {
+  constructor(tiff, poolOrDecoder, firstImage, omexmlString, offsets) {
+    this.poolOrDecoder = poolOrDecoder;
+    this.tiff = tiff;
+    this.type = 'ome-tiff';
```

Comment:
I think `VivViewerLayer` will accept any of these loaders, but I noticed the picture-in-picture viewer doesn't support the raster type 'bs tiff'. So technically the loaders are different depending on this? Might be worth returning something different in `getRaster` so that it works with all our viewers.

Reply:
Could you clarify here?  The error from zoom levels and `getRaster` is something I filed already.  I don't know what's going on but the legacy bioformats tiff `bf tiff` should in theory work with that.